fix: networkmanager starts when transport does not

NoelStephensUnity opened the pull request:
When the NetworkTransport.StartServer returns false, we need to stop the startup process for both the StartHost or StartServer methods in NetworkManager. Otherwise there are failure to bind error messages and the server or host still thinks it is started.
[MTT-2598](https://jira.unity3d.com/browse/MTT-2598)

## Changelog

### com.unity.netcode.gameobjects
-Fixed: issue where NetworkManager would continue starting if the NetworkTransport selected failed. 

## Testing and Documentation
* Includes integration tests.

NoelStephensUnity committed:
fix

This fixes the issue where if the Transport failed to start the NetworkManager would continue starting.

NoelStephensUnity committed:
test

additional validations to assure the appropriate NetworkManager(s) are not started after Transport failure detected.

NoelStephensUnity committed:
Update CHANGELOG.md

NoelStephensUnity committed:
update

removing override not being used.

NoelStephensUnity committed:
style

whitespace fix

NoelStephensUnity committed:
Merge branch 'develop' into fix/mtt-2598-networkmanager-starts-when-transport-does-not

NoelStephensUnity committed:
Merge branch 'develop' into fix/mtt-2598-networkmanager-starts-when-transport-does-not

NoelStephensUnity committed:
fix

merge issue with changelog

NoelStephensUnity committed:
fix

Need to assign a newly constructed ArraySegment for the FailedTransport.

NoelStephensUnity committed:
Merge branch 'develop' into fix/mtt-2598-networkmanager-starts-when-transport-does-not

0xFA11 commented on com.unity.netcode.gameobjects/CHANGELOG.md:
```diff
@@ -7,9 +7,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 ### Changed
+
 ### Fixed
+- Fixed issue where NetworkManager would continue starting if the NetworkTransport selected failed. (#1780)
```
```suggestion
- Fixed issue where NetworkManager would continue starting even if the NetworkTransport selected failed. (#1780)
```

0xFA11 commented on com.unity.netcode.gameobjects/Runtime/Core/NetworkManager.cs:
```diff
@@ -827,15 +827,24 @@ public bool StartServer()
-            IsServer = true;
-            IsClient = false;
-            IsListening = true;
+            var result = NetworkConfig.NetworkTransport.StartServer();
```
can we just kill `result` local variable entirely? (also below as well)
and turn if checks into `if (NetworkConfig.NetworkTransport.StartServer() == false)` etc.?

0xFA11 reviewed (approved):
minor suggestions but looks good otherwise

NoelStephensUnity committed:
Update com.unity.netcode.gameobjects/CHANGELOG.md

Co-authored-by: Fatih Mar <mfatihmar@gmail.com>

NoelStephensUnity committed:
update

This includes Fatih's suggestion to remove the transport returned Boolean variable.

NoelStephensUnity committed:
Merge branch 'develop' into fix/mtt-2598-networkmanager-starts-when-transport-does-not

netcode-ci-service committed:
Merge develop into fix/mtt-2598-networkmanager-starts-when-transport-does-not
